app/client/components/GridView.js

```diff
@@ -902,6 +904,24 @@ GridView.prototype.insertColumn = async function(colId = null, options = {}) {
   return newColInfo;
 };
 
+GridView.prototype.makeHeadersFromRow = async function(selection) {
+  if (this._getCellContextMenuOptions().disableMakeHeadersFromRow){
+    return;
+  }
+  const actions = this.viewSection.columns.peek().reduce((acc, col) => {
+    const colId = col.colId.peek();
+    let newColLabel = this.tableModel.tableData.getValue(selection.rowIds[0], colId);
+    if (Array.isArray(newColLabel) && newColLabel[0] === GristObjCode.List) {
+      newColLabel = newColLabel[1];
+    }
+    if (newColLabel) {
+      return [...acc, ['ModifyColumn', colId, {"label": newColLabel}]];
```

Comment:
`newColLabel` here is not always a string, right? I think we need the field's `formatter` here, which has a `formatAny` method that will produce an appropriate string.

For references specifically, there's some additional nuance because they display values from a different column. `ReferenceUtils.ts` could be useful for formatting those correctly.